### DISCUSSION TITLE
Octoprint Firmware Updater with Jyers Marlin?

### DISCUSSION THREAD

**Discussion by ryankesavan** [Jyers/Marlin#539] [Q&A] [answered]
Old:
Title : Is Marlin Binary File Transfer enabled? or any other way to use Octoprint Firmware Updater?
Is Marlin BFT enabled?
I am assuming no, since the changelog shows meatpack enabled and from what I can tell it is either one or the other.
I would like to use Octoprint Firmware Updater.

New (edit):
I'm a noob, apologies in advanced for errors. 
I haven't tried this myself yet because I am taking on too many tasks/projects and not getting things done.
Unless the precompiled firmware is updated to include either of the methods, you will have to do it on your own.

https://github.com/OctoPrint/OctoPrint-FirmwareUpdater/issues/190
This thread talks about the octoprint firmwareupdater plugin and Ender 3 v2. 
Posts about it working start around the middle to the end.
The firmware needs to be modified for either update method listed below.
If I'm not mistaken the preferred modification will be for the lpc176x method which should keep the ability to use meatpack.

https://github.com/OctoPrint/OctoPrint-FirmwareUpdate.
The processer on the creality 4.2.2/7 boards are apparently STM32 32-bit MCUs  (via SD card).
There are two flash methods (lpc176x or marlinbft) and both require modification to the firmware.
Guides for setting up either method:
https://github.com/OctoPrint/OctoPrint-FirmwareUpdater/blob/master/doc/lpc176x.md
https://github.com/OctoPrint/OctoPrint-FirmwareUpdater/blob/master/doc/marlinbft.md

I think the only other thing of importance for the plugin is to enable "Use timestamp filenames" under the advanced tab in the plugin settings.

**Comment by adam3654** [comment, marked as answer]
I just tried to set the plugin up (compiled fw removing meatpack, and adding BFT). The transfer seemed to work, but got a board reset failure error in octoprint. I pulled the sd card and there was a firmware file with the wrong date that was 0 bytes. I'm probably doing something wrong, but I can't find much help online. Let me know if you can get it to work for you.

**Reply by ryankesavan** [reply]
https://github.com/OctoPrint/OctoPrint-FirmwareUpdater/issues/190
someone mentioned a similar error in this

**Reply by adam3654** [reply]
> https://github.com/OctoPrint/OctoPrint-FirmwareUpdater/issues/190
> someone mentioned a similar error in this

After reviewing the linked thread, I checked my eeprom and it may have just worked! 

![Screenshot_20210420-151730.png](https://user-images.githubusercontent.com/16920869/115452078-f5e8e700-a1eb-11eb-8c05-0b564b6e6e8e.png)

I'll try it again with a goofy web address where the github.com/Jyers typically is, so I can M115 the info from the console after reset. 

BTW, I realized when manually updating via sd card, I can just fire off a M997 from octoprint to reset the board without power cycling the printer as I have my pi running off the printer's power supply. Octoprint just throws a warning about the board having been reset.

**Reply by darco415** [reply]
Its a shame the you have to choose between Meatpack and BFT....I chose Meatpack

**Reply by adam3654** [reply]
I just confirmed, it works on my machine! How cool.

**Reply by Jyers** [reply]
Yes it works, the board just doesn't report a reset. In FW updater there is a setting to stop the plugin from waiting for the reset response. I'd be very interested to try the lpc176x method, as it should be compatible with the current config settings.

**Reply by ryankesavan** [reply]
I quickly checked the merlin config files from the source code zip and it looks like `#define SDCARD_CONNECTION ONBOARD` is not setup in Configuration_adv.h file. I followed the rest of the instructions but when you click the test button in the plugin it gives an error. I think it said unable to write.

**Reply by Jyers** [reply]
`#define SDCARD_CONNECTION ONBOARD` shouldn't be required as ONBOARD should be the default, but I'll give it a try to see if enabling that helps.
